boards/arm/frdm_k64f/pinmux.c

```diff
@@ -35,81 +35,79 @@ static int frdm_k64f_pinmux_init(struct device *dev)
 
 #ifdef CONFIG_UART_MCUX_0
 	/* UART0 RX, TX */
-	pinmux_pin_set(portb, 16, PORT_PCR_MUX(kPORT_MuxAlt3));
-	pinmux_pin_set(portb, 17, PORT_PCR_MUX(kPORT_MuxAlt3));
+	pinmux_pin_set(portb, 16, kPORT_MuxAlt3);
+	pinmux_pin_set(portb, 17, kPORT_MuxAlt3);
 #endif
 
 #ifdef CONFIG_UART_MCUX_3
 	/* UART3 RX, TX */
-	pinmux_pin_set(portc, 16, PORT_PCR_MUX(kPORT_MuxAlt3));
-	pinmux_pin_set(portc, 17, PORT_PCR_MUX(kPORT_MuxAlt3));
+	pinmux_pin_set(portc, 16, kPORT_MuxAlt3);
+	pinmux_pin_set(portc, 17, kPORT_MuxAlt3);
 #endif
 	/* SW2 / FXOS8700 INT1 */
-	pinmux_pin_set(portc,  6, PORT_PCR_MUX(kPORT_MuxAsGpio));
+	pinmux_pin_set(portc,  6, kPORT_MuxAsGpio);
 
 	/* FXOS8700 INT2 */
-	pinmux_pin_set(portc, 13, PORT_PCR_MUX(kPORT_MuxAsGpio));
+	pinmux_pin_set(portc, 13, kPORT_MuxAsGpio);
 
 	/* SW3 */
-	pinmux_pin_set(porta,  4, PORT_PCR_MUX(kPORT_MuxAsGpio));
+	pinmux_pin_set(porta,  4, kPORT_MuxAsGpio);
 
 	/* Red, green, blue LEDs */
-	pinmux_pin_set(portb, 22, PORT_PCR_MUX(kPORT_MuxAsGpio));
-	pinmux_pin_set(porte, 26, PORT_PCR_MUX(kPORT_MuxAsGpio));
-	pinmux_pin_set(portb, 21, PORT_PCR_MUX(kPORT_MuxAsGpio));
+	pinmux_pin_set(portb, 22, kPORT_MuxAsGpio);
+	pinmux_pin_set(porte, 26, kPORT_MuxAsGpio);
+	pinmux_pin_set(portb, 21, kPORT_MuxAsGpio);
 
 #ifdef CONFIG_IEEE802154_MCR20A
 	/* FRDM-MCR20A Reset (D5) */
-	pinmux_pin_set(porta,  2, PORT_PCR_MUX(kPORT_MuxAsGpio));
+	pinmux_pin_set(porta,  2, kPORT_MuxAsGpio);
 	/* FRDM-MCR20A IRQ_B (D2) */
-	pinmux_pin_set(portb,  9, PORT_PCR_MUX(kPORT_MuxAsGpio));
+	pinmux_pin_set(portb,  9, kPORT_MuxAsGpio);
 #endif
 
 #ifdef CONFIG_SPI_0
 	/* SPI0 CS0, SCK, SOUT, SIN */
-	pinmux_pin_set(portd,  0, PORT_PCR_MUX(kPORT_MuxAlt2));
-	pinmux_pin_set(portd,  1, PORT_PCR_MUX(kPORT_MuxAlt2));
-	pinmux_pin_set(portd,  2, PORT_PCR_MUX(kPORT_MuxAlt2));
-	pinmux_pin_set(portd,  3, PORT_PCR_MUX(kPORT_MuxAlt2));
+	pinmux_pin_set(portd,  0, kPORT_MuxAlt2);
+	pinmux_pin_set(portd,  1, kPORT_MuxAlt2);
+	pinmux_pin_set(portd,  2, kPORT_MuxAlt2);
+	pinmux_pin_set(portd,  3, kPORT_MuxAlt2);
 #endif
 
 #if CONFIG_I2C_0
 	/* I2C0 SCL, SDA */
-	pinmux_pin_set(porte, 24, PORT_PCR_MUX(kPORT_MuxAlt5)
-					| PORT_PCR_ODE_MASK);
-	pinmux_pin_set(porte, 25, PORT_PCR_MUX(kPORT_MuxAlt5)
-					| PORT_PCR_ODE_MASK);
+	pinmux_pin_set(porte, 24, kPORT_MuxAlt5 | PORT_PCR_ODE_MASK);
+	pinmux_pin_set(porte, 25, kPORT_MuxAlt5 | PORT_PCR_ODE_MASK);
```

Comment:
This won't work whenever we need to set bits outside the MUX field, in this case ODE.

Reply:
@MaureenHelm: Ho right! 'Find all/replace all/Don't think'...
So we need to have all bindings available in include/dt_bindings with a constraint that dt parser only accept basic #define. enum/struct/.. could not be used.
Providing a new version using set of bindings based on PORT_PCR...

Reply:
@MaureenHelm : Happy new year and best wishes to you!
Would you mind have a new look to this PR and check if you have some more comments?